File: Employee-Management-Systems/src/main/java/com/example/EmployeeManagementSystems/services/EmployeeService.java
package com.example.EmployeeManagementSystems.services;

import com.example.EmployeeManagementSystems.entities.Employee;

import java.util.List;

public interface EmployeeService {
    Employee addEmployee(Employee employee);
    Employee updateEmployee(Employee employee, Long id);
    List<Employee> getAllEmployees();
    Employee getEmployeeById(Long id);
    void deleteEmployee(Long id);

}

File: FrontEnd_Employee_Management_System/src/app/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Employee } from './employee';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  private baseUrl= "http://localhost:8080/employees";
  constructor( private httpClient: HttpClient ) { }

  getEmployeesList(): Observable<Employee[]>{
    return this.httpClient.get<Employee[]>(`${this.baseUrl}`);
  }

  addEmployee(employee:Employee): Observable<Object> {
    return this.httpClient.post(`${this.baseUrl+"/add"}`,employee)
  }

  updateEmployee(id: number, employee:Employee): Observable<Object>{
      return this.httpClient.put(`${this.baseUrl+"/add"}/${id}`, employee);
  }

  getEmployeeById(id: number):Observable<Employee>{
    return this.httpClient.get<Employee>(`${this.baseUrl}/${id}`);
  }

  deleteEmployee(id:number):Observable<Object>{
    return this.httpClient.delete(`${this.baseUrl+"/delete"}/${id}`);
  }
}
